notebooks/2018-10-05-create-netcdf.ipynb

```diff
@@ -0,0 +1,368 @@
+   "metadata": {},
+   "source": [
+    "## Result\n",
+    "\n",
+    "Looking at the output file, we notice something:\n",
+    "\n",
+    "```\n",
+    "$ ncdump -h data.nc\n",
+    "netcdf data {\n",
+    "dimensions:\n",
+    "\tlongitude = 200 ;\n",
+    "\tlatitude = 101 ;\n",
+    "variables:\n",
+    "\tfloat longitude(longitude) ;\n",
+    "\t\tlongitude:units = \"degree_east\" ;\n",
+    "\tfloat latitude(latitude) ;\n",
+    "\t\tlatitude:units = \"degree_north\" ;\n",
+    "\tfloat field(latitude, longitude) ;\n",
+    "\t\tfield:_FillValue = -2.e+08f ;\n",
+    "\n",
+    "// global attributes:\n",
+    "\t\t:note = \"training file created with Fortran 90\" ;\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "In Fortran, we always had the dimensions of `field` as (longitude, latitude), now it's the other way round.\n",
+    "\n",
+    "That's because Fortran stores multi-dimensional arrays the other way as almost all other programming languages."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Python\n",
+    "----\n",
+    "\n",
+    "In python, things are a lot easier. It can store the complete datastructure in memory, the output is then only a single line of code:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, import `xarray` and `numpy`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import xarray as xr\n",
+    "import numpy as np\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Second: Create the latitude and longitude arrays:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nx = 200; ny = 101\n",
+    "lon_array = np.linspace(0, 360, nx, endpoint=False, dtype=np.float32)\n",
+    "lat_array = np.linspace(-90, 90, ny, endpoint=True, dtype=np.float32)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Third: Create the field, complete with values, dimensions, coordinates, and attributes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "field=xr.DataArray(\n",
+    "    np.sin(lon_array[np.newaxis, :] * np.pi / 180.) * \n",
+    "    np.cos(lat_array[:, np.newaxis] * np.pi / 180.),\n",
+    "    dims = ['latitude', 'longitude'], \n",
+    "    coords = {'longitude': lon_array, 'latitude': lat_array},\n",
+    "    attrs = {'_FillValue':-2e8},\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Fourth: The coordinate arrays don't have units yet, let's fix that."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "field.latitude.attrs = {'units':'degree_north'}\n",
```

Comment:
This overwrites existing attributes, prefer `field.latitude.attrs['units'] = 'degrees_north'`